v1/lists.js

```diff
@@ -0,0 +1,190 @@
+'use strict';
+
+const P = require('bluebird');
+const preq = require('preq');
+const HyperSwitch = require('hyperswitch');
+const URI = HyperSwitch.URI;
+const spec = HyperSwitch.utils.loadSpec(`${__dirname}/lists.yaml`);
+
+class ReadingLists {
+    /**
+     * @param {!Object} options RESTBase options object.
+     *   Apart from standard options, the following are recognized:
+     *   - summaryConcurrency: the max number of parallel requests to summary endpoints (default: 10)
+     */
+    constructor(options) {
+        this.options = options;
+    }
+
+    /**
+     * Transform the continuation data into a string so it is easier for clients to deal with.
+     * @param {!Object|undefined} continuation Continuation object returned by the MediaWiki API.
+     * @return {!String|undefined} Continuation string.
+     */
+    flattenContinuation(continuation) {
+        return JSON.stringify(continuation);
+    }
+
+    /**
+     * Inverse of flattenContinuation.
+     * @param {!String} continuation Continuation string returned by flattenContinuation().
+     * @return {!Object|undefined} Continuation object.
+     */
+    unflattenContinuation(continuation) {
+        const sanitizedContinuation = {};
+        if (typeof continuation === 'string') {
+            try {
+                continuation = JSON.parse(continuation);
+            } catch (e) {
+                this.options.log('error/unflatten', {
```

Comment:
It doesn't log any details though. I guess the stack trace is pretty pointless but the raw json and the error message from the parser are helpful for debugging.

Eventually I'd like to use `mwUtil.encodePagingToken` or something similar so that users cannot trigger any error-level logging here, but I considered that less important than getting the service set up on beta.